core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/UpdateViewHook.java

```diff
@@ -0,0 +1,317 @@
+                if (null != originalView) {
+                    originalView.expandGlobalDefinitions();
+                    if ((null != whiteListElementGroups && !whiteListElementGroups.isEmpty())
+                            || (null != blackListElementGroups && !blackListElementGroups.isEmpty())) {
+
+                        Map<String, ViewElementDefinition> entities = Maps.newHashMap(originalView.getEntities());
+                        Set<Entry<String, ViewElementDefinition>> entitiesEntrySet = entities.entrySet();
+                        entitiesEntrySet.removeIf(this::removeElementGroups);
+                        viewBuilder.entities(entities);
+
+                        Map<String, ViewElementDefinition> edges = Maps.newHashMap(originalView.getEdges());
+                        Set<Entry<String, ViewElementDefinition>> edgesEntrySet = edges.entrySet();
+                        edgesEntrySet.removeIf(this::removeElementGroups);
+                        viewBuilder.edges(edges);
+                    }
+                }
+                operationView.setView(viewBuilder.build());
+            }
+        }
+    }
+
+    protected final View.Builder mergeView(final OperationView operationView, final View viewToMerge) {
+        View.Builder viewBuilder = new View.Builder()
+                .merge(operationView.getView());
+
+        if (null != viewToMerge) {
+            viewBuilder.merge(viewToMerge.clone());
+        }
+
+        return viewBuilder;
+    }
+
+    protected final boolean removeElementGroups(final Entry<String, ViewElementDefinition> entry) {
+        boolean remove;
+
+        if (null != whiteListElementGroups) {
+            remove = !whiteListElementGroups.contains(entry.getKey());
+        } else {
+            remove = false;
+        }
+
+        if (!remove) {
+            if (null != blackListElementGroups) {
+                remove = blackListElementGroups.contains(entry.getKey());
+            }
+        }
+        return remove;
+    }
+
+    protected final boolean applyToUser(final User user) {
+        boolean userHasWithDataAuth = validateAuths(user.getDataAuths(), this.withDataAuth, true);
+        boolean userHasWithOpAuth = validateAuths(user.getOpAuths(), this.withOpAuth, true);
+        boolean userHasWithoutDataAuth = validateAuths(user.getDataAuths(), this.withoutDataAuth, false);
+        boolean userHasWithoutOpAuth = validateAuths(user.getOpAuths(), this.withoutOpAuth, false);
+
+        return userHasWithDataAuth && userHasWithOpAuth && !userHasWithoutDataAuth && !userHasWithoutOpAuth;
+    }
+
+    protected final boolean validateAuths(final Set<String> userAuths, final Set<String> validAuth, final boolean ifValidAuthIsNull) {
+        boolean rtn = ifValidAuthIsNull;
+        if (null != validAuth) {
+            if (null == userAuths) {
+                rtn = validAuth.isEmpty();
+            } else {
+                HashSet<String> temp = Sets.newHashSet(validAuth);
```

Comment:
Rather than creating a new HashSet every time, you could use something like CollectionUtils.containsAny(userAuths, validAuth). This just iterates around the collections.

Reply:
👍